src/dmd/e2ir.d

```diff
@@ -2224,7 +2224,7 @@ elem *toElem(Expression e, IRState *irs)
                     foreach (i; 1 .. depth - d)
                         e1 = (cast(CastExp)e1).e1;
 
-                    el = toElemCast(cast(CastExp)e1, el);
+                    el = toElemCast(cast(CastExp)e1, el, true);
```

Comment:
An enum as the last parameter would make this readable at call site.

Reply:
Named parameters is better.

Reply:
?? D doesn't have those...

Reply:
it will

Reply:
Even if it does, it'll be years before we can use it in DMD due to bootstrapping. That code is being written / read now.

Reply:
An enum would be better, but let's not get stalled on minor details. Dmd does not enforce this style and we can always make another refactoring PR to change this.

Reply:
Unfortunately "minor details" lead to one big hard to maintain blob in the long run - and then less and less want to work with the blob. Refactoring rarely happens. And it probably takes less time to "fix" a minor detail like this than to rationalize to yourself why not to. But whatevs :)

Reply:
You are right, but instead of waiting for Walter to due the changes, it is faster to merge and open a new PR. The change is trivial and anyone can do it (including you) right away.

Reply:
True, it's trivial. But I'm not interested in cleaning up minor details after people if they don't care enough about their own projects (specifically if it's pointed out). And I don't think many people are.